lib/logstash/filters/jdbc_static.rb

```diff
@@ -0,0 +1,186 @@
+#
+# [source,ruby]
+# filter {
+#   jdbc_static {
+#     jdbc_driver_library => "/path/to/mysql-connector-java-5.1.34-bin.jar"
+#     jdbc_driver_class => "com.mysql.jdbc.Driver"
+#     jdbc_connection_string => ""jdbc:mysql://localhost:3306/mydatabase"
+#     jdbc_user => "me"
+#     jdbc_password => "secret"
+#     statement => "select * from WORLD.COUNTRY WHERE Code = :code"
+#     parameters => { "code" => "country_code"}
+#     target => "country_details"
+#   }
+# }
+#
+module LogStash module Filters class JdbcStatic < LogStash::Filters::Base
+  config_name "jdbc_static"
+
+  # Define the set (Hash) of loaders to fetch remote data and create local tables.
+  # the fetched data will be inserted into the local tables. Make sure that the
+  # local table name, columns and datatypes correspond to the shape of the remote data
+  # being fetched.
+  # You can also specify alternate JDBC strings to use if there is more than one remote database
+  # For example:
+  # loaders => {
+  #   "country_details" => {
+  #     "query" => "select code, name from WORLD.COUNTRY",
+  #     "max_rows" => 2000
+  #   },
+  #   "servers" => {
+  #     "jdbc_driver_class" => "org.postgresql.Driver",
+  #     "jdbc_connection_string" => "jdbc:postgres://user:password@remotedb/infra"
+  #     "query" => "select ip, name, location from INTERNAL.SERVERS",
+  #   }
+  # }
+  config :loaders, :validate => :hash, :required => true
+
+  # Define an array of SQL DDL statements to execute when the plugin first starts.
+  # These will usually be the creational SQL to setup the local in-memory tables.
+  # For example:
+  # before_load_sql => [
+  #   "CREATE TABLE country (code VARCHAR(3) NOT NULL, name VARCHAR(32) NOT NULL)",
+  #   "CREATE TABLE servers (ip VARCHAR(16) NOT NULL, name VARCHAR(32) NOT NULL, location VARCHAR(32) NOT NULL)",
+  #   "CREATE INDEX country_code_index ON country(code)",
+  #   "CREATE INDEX servers_ip_index ON servers(ip)"
+  # ]
+  # NOTE: Important! Tables created here must have the same names as those used in the `loaders` and
+  # `lookups` configuration options
+  config :before_load_sql, :validate => :array, :default => []
+
+  # Define an array of SQL DDL statements to execute after the remote data is fetched
+  # and inserted locally. If the CRON schedule is specifed, the remote data is periodically
+  # fetched and so these SQL statements will be executed repeatedly.
+  config :after_load_sql, :validate => :array, :default => []
+
+  # Define the set (Hash) of enhancement lookups to be applied to an event
+  # The key is the target field and the value is a hash of the query string and an optional
+  # parameter hash. Target is overwritten if existing.
+  # Use parameters to have this plugin put values from the event into the query.
+  # For example:
+  # lookups => {
+  #   "country_details" => {
+  #     "query" => "select * from country WHERE code = :code",
+  #     "parameters" => {"code" => "country_code"}
```

Comment:
We should add a "target" field here for specification of the field destination of the lookup.

- By default (if not specified), it will bring all looked up fields to the top level. If there are field conflicts, it will overwrite.
- If target is specified (one field allowed), then it will store all looked up fields in that key as a nested object.

@guyboertje @jordansissel